probe/probe.go

```diff
@@ -140,20 +152,33 @@ func (p *Probe) tag(r report.Report) report.Report {
 	return r
 }
 
+func (p *Probe) drainAndPublish(rpt report.Report, rs chan report.Report) {
+ForLoop:
+	for {
+		select {
+		case r := <-rs:
+			rpt = rpt.Merge(r)
+		default:
+			break ForLoop
+		}
+	}
+
+	if err := p.publisher.Publish(rpt); err != nil {
+		log.Printf("publish: %v", err)
+	}
+}
+
 func (p *Probe) publishLoop() {
 	defer p.done.Done()
-	var (
-		pubTick = time.Tick(p.publishInterval)
-		rptPub  = xfer.NewReportPublisher(p.publisher)
-	)
+	pubTick := time.Tick(p.publishInterval)
 
 	for {
 		select {
 		case <-pubTick:
-			localReport := p.rpt.swap(report.MakeReport())
-			if err := rptPub.Publish(localReport); err != nil {
-				log.Printf("publish: %v", err)
-			}
+			p.drainAndPublish(report.MakeReport(), p.spiedReports)
+
+		case rpt := <-p.shortcutReports:
+			p.drainAndPublish(rpt, p.shortcutReports)
```

Comment:
It will do exactly that.  Luckily the reports are CRDTs, so merging them together will always result in the same output, independent of order.